lib/streamlit/runtime/caching/hashing.py

```diff
@@ -291,6 +288,11 @@ def _to_bytes(self, obj: Any) -> bytes:
 
             self.update(h, obj.tobytes())
             return h.digest()
+        elif type_util.is_type(obj, "PIL.Image.Image"):
+            import numpy as np
+
+            np_array = np.frombuffer(obj.tobytes(), dtype="uint8")
```

Comment:
```python
np_array = np.frombuffer(obj.tobytes(), dtype="uint8")
``` 
is ~2 times faster than naive 
```python
np_array  = np.array(obj)
``` 
or even

```python
np_array  = np.array(obj, copy=False)
``` since copy happens anyway because of the internal structure of PIL image because it is tuples of integers, not flat integers)

https://replit.com/join/onbduhvghk-kajarenc

Reply:
Can we just hash the value of `obj.tobytes()` (the value you pass in to `np.frombuffer`?)